### ISSUE TITLE
incorrect fold

### ISSUE THREAD

**Issue by kzhui125** [zed-industries/zed#11495]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

```rs
fn main() {
    let str = r"1
2
3
4
5";

    println!("{}", str);
}
```

before fold(VSCode and Zed):

<img width="555" alt="image" src="https://github.com/zed-industries/zed/assets/11925432/f9e850c1-91a8-4a2b-be76-68c66ed4ead2">

after fold(VSCode and Zed):

<img width="566" alt="image" src="https://github.com/zed-industries/zed/assets/11925432/22adc3a3-5b8e-47a7-9489-db22857932b9">

### Environment

Zed: v0.133.7 (Zed)
OS: macOS 14.4.1
Memory: 32 GiB
Architecture: aarch64

**Comment by sineptic** [comment]
I can reproduce it on 0.184.8

**Comment by sineptic** [comment]
but I think it is duplicate of #5057

**Comment by MrSubidubi** [comment]
Yeah it definitely is, thanks for the heads-up @sineptic . And sorry this is still happening. There is definitely much room for improvement on that end, but it will require a lot of rewriting and some more changes. See https://github.com/zed-industries/zed/issues/22703 for example.
